sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -986,11 +996,13 @@ public boolean advance() throws IOException {
               rawRecord.topic(),
               rawRecord.partition(),
               rawRecord.offset(),
+              consumerSpEL.getEventTimestamp(rawRecord),
               decode(rawRecord.key(), source.spec.getKeyCoder()),
               decode(rawRecord.value(), source.spec.getValueCoder()));
 
           curTimestamp = (source.spec.getTimestampFn() == null)
```

Comment:
curTimestamp is same as record timestamp. Here, we apply user function only to curTimestamp, why? To avoid this confusion, I suggest removing curTimestamp.